src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -464,6 +464,104 @@ public void It_doesnt_includes_requires_preview_features()
             Assert.False(contains);
         }
 
+        [Theory]
+        [InlineData(true, true, "net6.0", false)]
+        [InlineData(true, true, "net7.0", true)]
+        [InlineData(true, true, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(true, false, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(false, false, ToolsetInfo.CurrentTargetFramework, true)]
+        public void TestDisableRuntimeMarshalling(bool disableRuntimeMarshalling, bool generateDisableRuntimeMarshallingAttribute, string targetFramework, bool shouldHaveAttribute)
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld", identifier: $"{disableRuntimeMarshalling}${generateDisableRuntimeMarshallingAttribute}${targetFramework}")
+                .WithSource()
+                .WithTargetFramework(targetFramework)
+                .WithProjectChanges((path, project) =>
+                {
+                    var ns = project.Root.Name.Namespace;
+
+                    project.Root.Add(
+                        new XElement(ns + "PropertyGroup",
+                            new XElement(ns + "DisableRuntimeMarshalling", $"{disableRuntimeMarshalling}")));
+
+                    if (disableRuntimeMarshalling && !generateDisableRuntimeMarshallingAttribute)
+                    {
+                        project.Root.Add(
+                            new XElement(ns + "PropertyGroup",
+                                new XElement(ns + "GenerateDisableRuntimeMarshallingAttribute", $"False")));
+                    }
+                });
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute().Should().Pass();
+
+            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory(targetFramework).FullName, "HelloWorld.dll");
+
+            var parameterlessAttributes = AssemblyInfo.GetParameterlessAttributes(assemblyPath);
+            bool contains = false;
+            foreach (var attribute in parameterlessAttributes)
+            {
+                if (attribute.Equals("DisableRuntimeMarshallingAttribute", System.StringComparison.Ordinal))
+                {
+                    contains = true;
+                    break;
+                }
+            }
+
+            if (disableRuntimeMarshalling && generateDisableRuntimeMarshallingAttribute)
+            {
+                if (shouldHaveAttribute)
+                {
+                    Assert.True(contains);
+                }
+                else
+                {
+                    // The assembly level attribute is generated only for .NET 7 and newer
+                    Assert.False(contains);
+                }
+            }
+
+            if (!generateDisableRuntimeMarshallingAttribute)
+            {
+                Assert.False(contains);
+            }
```

Comment:
I think you express whether the attribute should be generated in the theory parameters, so you can simplify the logic by only checking that value to see what you should assert.

```suggestion
            if (shouldHaveAttribute)
            {
                Assert.True(contains);
            }
            else
            {
                // The assembly level attribute is generated only for .NET 7 and newer
                Assert.False(contains);
            }
```